Libraries/Blob/RCTBlobCollector.mm

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "RCTBlobCollector.h"
+
+#import <React/RCTBridge+Private.h>
+#import "RCTBlobManager.h"
+
+namespace facebook {
+namespace react {
+
+RCTBlobCollector::RCTBlobCollector(RCTBlobManager *blobManager, const std::string &blobId)
+: blobId_(blobId), blobManager_(blobManager) {}
+
+RCTBlobCollector::~RCTBlobCollector() {
+  RCTBlobManager *blobManager = blobManager_;
+  NSString *blobId = [NSString stringWithUTF8String:blobId_.c_str()];
+  dispatch_async([blobManager_ methodQueue], ^{
+    [blobManager remove:blobId];
+  });
+}
+
+void RCTBlobCollector::install(RCTBlobManager *blobManager) {
+  RCTCxxBridge *cxxBridge = (RCTCxxBridge *)blobManager.bridge;
+  if (cxxBridge.runtime == nullptr) {
```

Comment:
Looking at this code again I can probably add a bit more safety: use a `__weak` ref for `cxxBrige` and move the `cxxBridge.runtime == nullptr` inside the dispatched block.